### PR TITLE
fix: slugify from creating double+ dashes 

### LINKED ISSUE #13980: [CAL-3260] Double & Triple dashes getting in to public booking links
We have had two cases recently of users ending up with "--" and "---" in their public booking links and when you try to visit the page it is throwing a 404.

In both cases, the users had space, -, space or similar with a (. When you type these out when creating an event type you can see that we add double and triple dashes in the URL in the modal but after you hit proceed we strip it out.

I think when users are updating their username, we are regenerating the links and taking the URL from the title again but not stripping it out.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3260](https://linear.app/calcom/issue/CAL-3260/double-and-triple-dashes-getting-in-to-public-booking-links)</sub>

### PR DESCRIPTION
**Pull request by sean-brydon**
fixes: #13980 

Before:

"Hello -  World 123_ !@#  Test    456   789" -> "hello---world-123----test-456-789"
Now
"Hello -  World 123_ !@#  Test    456   789" ->""hello-world-123-test-456-789"

### CODE REVIEW

File: packages/lib/slugify.test.ts [hunk 1]
```diff
@@ -45,6 +45,10 @@ describe("slugify", () => {
     expect(slugify("hellothere.")).toEqual("hellothere");
   });
 
+  it("Should replace consecutive dashes with a single dash", () => {
+    expect(slugify("Hello -  World 123_ !@#  Test    456   789")).toEqual("hello-world-123-test-456-789");
+  });
+
   // This is failing, if we want to fix it, one approach is as used in getValidRhfFieldName
   it.skip("should remove unicode and emoji characters", () => {
     expect(slugify("Hello 📚🕯️®️ There")).toEqual("hello---------there");
```
Comment (Udit-takkar) [inline packages/lib/slugify.test.ts:54]: this old test case too many dashes in expected value

Comment (sean-brydon) [reply packages/lib/slugify.test.ts:54]: Oh i missed that as its being skipped let me see if i can fix the test 

No this new slugify will only allow one dash in a row

**Review by Udit-takkar — APPROVED** [review]

**Comment by PeerRich** [comment]
i was just about to open an issue for this lol. great work!

**Comment by PeerRich** [comment]
tests seem flaky https://github.com/calcom/cal.com/actions/runs/8188418835/job/22391277351?pr=14002

**Pull request merged**